lib/view_component/slotable_v2.rb

```diff
@@ -227,7 +227,9 @@ def set_slot(slot_name, *args, **kwargs, &block)
         # current component. This is necessary to allow the lambda to access helper
         # methods like `content_tag` as well as parent component state.
         renderable_value = if block_given?
-          slot_definition[:renderable_function].bind(self).call(*args, **kwargs) { view_context.capture(&block) }
+          slot_definition[:renderable_function].bind(self).call(*args, **kwargs) do |*args, **kwargs|
+            view_context.capture { block.call(*args, **kwargs) }
```

Comment:
```suggestion
            view_context.capture(*args, **kwargs, &block)
```

Related to https://github.com/github/view_component/pull/683, I think this will need to be changed to prevent breaking HAML templates.